File: 0001-TwoSum/TwoSum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int [] response = new int [2];
        Map<Integer, Integer> arrayMap = new HashMap<>();
        for(Integer index = 0; index < nums.length; index += 1) {
            int needleValue = target - nums[index];
            if(arrayMap.containsKey(needleValue)) {
                int needleKey = arrayMap.get(needleValue);
                if(needleKey != index) {
                    response[0] = needleKey;
                    response[1] = index;
                    return response;
                }
           }
            arrayMap.put(nums[index], index);
        }
        return response;
    }
}

File: 0001-TwoSum/TwoSum.php
<?php
/**
 * author: yanyandenuonuo
 * create: 2018.05.13
 */

$array = [1, 2, 3, 4, 5, 6, 7, 8, 9];


/**
 * @param $nums
 * @param $target
 * @return array
 */
function twoSum($nums, $target) {
    for($index = 0; $index < count($nums); $index += 1) {
        $key = array_search($nums, $target - $nums[$index]);
        if($key !== false && $key != $index) {
            return [$index, $key];
        }
    }
}


var_dump(twoSum($array, 9));
